### ISSUE TITLE
Error when installing Arch Linux rootfs on Windows 10 Fall Creators Update

### ISSUE THREAD

**Issue by Calinou** [RoliSoft/WSL-Distribution-Switcher#69]
With Windows 10 Fall Creators Update + a fresh Bash on Windows install from the Microsoft Store, I use these commands:

```
python3 get-prebuilt.py base/archlinux
python3 install.py rootfs_base_archlinux_latest.tar.gz
```

It results in:

```
[*] Probing the Linux subsystem...
[!] Failed to get home directory of default user in WSL: [WinError 2] The system cannot find the file specified
```

**Comment by BonusPlay** [comment]
Same here, however mine doesn't work before or after fall creator's update.

**Comment by AnneTheAgile** [comment]
@Calinou not sure if related, but some users have had some trouble with not finding files. The latest creators update is supposed to fix some issues but possibly more eg, this one remain; https://github.com/Microsoft/WSL/issues/2712
Do you think it is related?

**Comment by BonusPlay** [comment]
Indeed, updating helped, I'm currently on 16299.98 and it's working. I had same problem as described in #64, but fix provided there helped. However I have noticed some strange bugs, like backspace placing a space instead of deleting last character or missing first character from line (`oot@Zyga4:/#` is kinda missing 'r'), but I believe that is not related to this project.

**Comment by MatheusGrilo** [comment]
It was actually moved in the latest release to :

> C:\Users\%USERNAME%\AppData\Local\Packages\CanonicalGroupLimited.UbuntuonWindows_79rhkp1fndgsc\LocalState\rootfs\

Source: https://stackoverflow.com/questions/39886862/the-lxss-folder-in-windows-for-bash-on-ubuntu-on-windows-is-missing

That's why u can't install, because it tries to clone into 

> C:\Users\%USERNAME%\AppData\Local\lxss

**Comment by BonusPlay** [comment]
Not true, folder/symlink `C:\Users\%USERNAME%\AppData\Local\lxss` is still there, just hidden. You just have to uncheck this.
![image](https://user-images.githubusercontent.com/8405359/33804784-ba590b5e-ddac-11e7-9a86-3bdf7fd95e65.png)
Installation works just fine, but you have to update your windows.

**Comment by MatheusGrilo** [comment]
Not here, it just don't exist (it's already unchecked). Which Windows version are u using?

![https://i.imgur.com/qKS9gk3.jpg](https://i.imgur.com/qKS9gk3.jpg)

![https://i.imgur.com/0hfxyaU.jpg](https://i.imgur.com/0hfxyaU.jpg)

![https://i.imgur.com/lZYPPNA.jpg](https://i.imgur.com/lZYPPNA.jpg)

**Comment by BonusPlay** [comment]
Windows 10 Pro, version 1709, build 16299.98
![image](https://user-images.githubusercontent.com/8405359/33805481-f2886800-ddb9-11e7-8211-0c4abbe3f7ee.png)

**Comment by AngellusMortis** [comment]
lxrun has been deprecated, it will likely be removed in the next version of Windows. If you are on a fresh install of Fall Creators and install Ubuntu according to the new set up, it does not use lxrun.

```
Warning: lxrun.exe is only used to configure the legacy Windows Subsystem for Linux distribution.
Distributions can be installed by visiting the Windows Store:
https://aka.ms/wslstore

Usage:
    /install - Installs the subsystem
        Optional arguments:
            /y - Do not prompt user to accept or create a subsystem user
    /uninstall - Uninstalls the subsystem
        Optional arguments:
            /full - Perform a full uninstall
            /y - Do not prompt user to confirm
    /setdefaultuser - Sets the default subsystem user. If the user account does not exist it will be created.
        Optional arguments:
            username - Supply the username
            /y - If username is supplied, do not prompt to create a password
```

**Comment by DuIslingr** [comment]
@BonusPlay the reason its still there is for you is because you had it installed prior to fall creators update. a clean install will place it in the new location and no longer uses lxss.

**Comment by brianneve** [comment]
@RoliSoft 
Same issue [!] Failed to get home directory of default user in WSL: [WinError 2] here.
Please help. Build 16299.125

**Comment by imag0r** [comment]
Workaround that worked for me: Uninstall the distribution you got from Windows Store (from App/Remove Programs), then type "lxrun /install" command to re-install it.

**Comment by Luckz** [comment]
https://github.com/izolight/WSL-Distribution-Switcher claims to have "fixes for Arch", maybe that fork works better?

**Comment by berney** [comment]
@Luckz I haven't tested it, but just comparing the source I don't think it will fix the Windows 10 Falls Creator Update compatibility issue.
